lib/workload/stateless/stacks/icav1-copy-batch-utility/lambdas/lambda.py

```diff
@@ -0,0 +1,184 @@
+            if tagging_copy == 'Enable':
+                get_obj_tag = s3Client.get_object_tagging(Bucket=s3Bucket, Key=s3Key)
+
+        # Construct New Path
+        # Construct New Key
+        if new_prefix and len(new_prefix) > 0:
+            newKey = "{0}/{1}".format(new_prefix, s3Key)
+        else:
+            newKey = s3Key
+
+        newBucket = target_bucket
+
+        # Toggle Metadata or Tagging Copy Based on Enviromental Variables
+        # Construct Request Parameters with metadata and tagging from sourceKey
+        # Create variables to append as metadata and tagging to destination object
+        if metadata_copy == 'Enable':
+            logger.info("Object Metadata Copy Enabled from Source to Destination")
+            cache_control = get_metadata.get('CacheControl')
+            content_disposition = get_metadata.get('ContentDisposition')
+            content_encoding = get_metadata.get('ContentEncoding')
+            content_language = get_metadata.get('ContentLanguage')
+            metadata = get_metadata.get('Metadata')
+            website_redirect_location = get_metadata.get('WebsiteRedirectLocation')
+            expires = get_metadata.get('Expires')
+            # Construct Request With Required and Available Arguments
+            if cache_control:
+                myargs['CacheControl'] = cache_control
+            if content_disposition:
+                myargs['ContentDisposition'] = content_disposition
+            if content_encoding:
+                myargs['ContentEncoding'] = content_encoding
+            if content_language:
+                myargs['ContentLanguage'] = content_language
+            if metadata:
+                myargs['Metadata'] = metadata
+            if website_redirect_location:
+                myargs['WebsiteRedirectLocation'] = website_redirect_location
+            if expires:
+                myargs['Expires'] = expires
+        else:
+            logger.info("Object Metadata Copy Disabled")
+
+        if tagging_copy == 'Enable':
+            logger.info("Object Tagging Copy Enabled from Source to Destination")
+            existing_tag_set = (get_obj_tag.get('TagSet'))
+            # Convert the Output from get object tagging to be compatible with transfer s3.copy()
+            tagging_to_s3 = "&".join([f"{parse.quote_plus(d['Key'])}={parse.quote_plus(d['Value'])}" for d in existing_tag_set])
+            # Construct Request With Required and Available Arguments
+            if existing_tag_set:
+                myargs['Tagging'] = tagging_to_s3
+        else:
+            logger.info("Object Tagging Copy Disabled")
+
+        # Initiate the Actual Copy Operation and include transfer config option
+        logger.info(f"starting copy of object {s3Key} with versionID {s3VersionId} between SOURCEBUCKET: {s3Bucket} and DESTINATIONBUCKET: {newBucket}")
+        response = s3Client.copy(copy_source, newBucket, newKey, Config=transfer_config, ExtraArgs=myargs)
```

Comment:
How long would this be expected to take for a 100 Gb file?

Reply:
We'll see after implementing this: https://aws.amazon.com/blogs/storage/copying-objects-greater-than-5-gb-with-amazon-s3-batch-operations/